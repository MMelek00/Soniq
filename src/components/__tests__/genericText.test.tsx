/**
 * @format
 */

import 'react-native';
import React from 'react';

import renderer from 'react-test-renderer';
import GenericText, {TextType} from '../genericText';
describe('GenericText', () => {
  it('renders correctly with Error', () => {
    const tree = renderer.create(
      <GenericText
        type={TextType.Error}
        size={32}
        text="An error occurred..."
      />,
    );
    expect(tree).toMatchSnapshot();
  });
  it('renders correctly with Normal Text', () => {
    const tree = renderer.create(
      <GenericText type={TextType.Normal} size={32} text="Normal Text" />,
    );
    expect(tree).toMatchSnapshot();
  });
});
