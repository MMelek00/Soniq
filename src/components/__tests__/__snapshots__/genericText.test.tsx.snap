// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericText renders correctly with Error 1`] = `
<Text
  size={32}
  style={
    [
      {
        "color": "red",
        "fontSize": 32,
        "fontWeight": "normal",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "textalign": "'center'",
      },
    ]
  }
  type="error"
>
  An error occurred...
</Text>
`;

exports[`GenericText renders correctly with Normal Text 1`] = `
<Text
  size={32}
  style={
    [
      {
        "color": "black",
        "fontSize": 32,
        "fontWeight": "normal",
        "marginBottom": 10,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 10,
        "textalign": "'center'",
      },
    ]
  }
  type="normal"
>
  Normal Text
</Text>
`;
